tests/test_format_alerts.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+# -*- coding: UTF-8 -*-
+
+"""This file provides tests for the alert formatting performed prior to storage
+in GCS. This includes testing the generation of the valid schema pickle file by
+the ``alert_ingestion.valid_schemas.gen_valid_schema`` module, and the
+correction of alert schemas by the ``alert_ingestion.consume`` module.
+
+To test a new survey version, store the Avro files for a few alerts in the
+``test_alerts`` directory. Include the name and version of the survey in the
+file name, separated by underscores. For the version, prepend a "v" and replace
+the period with a dash. For example, the filename for a ZTF version 3.3 alert
+should be: ``ztf_v3-3_originalfilename.avro``.
+"""
+
+from pathlib import Path
+from unittest import TestCase
+
+from broker.alert_ingestion import consume
+from broker.alert_ingestion.valid_schemas.gen_valid_schema import _load_Avro
+
+test_alerts_dir = Path(__file__).parent / 'test_alerts'
+
+
+# Quick thoughts on test outlines for alert formatting
+class AlertFormattingDataUnchanged(TestCase):
+    """Test data we retrieve from file before / after formatting is the same"""
+
+    def get_survey_and_schema(self, filename: str) -> (str, float):
+        """ Parses the filename to get the survey and version"""
+        f = filename.split('_')
+        survey = f[0]
+        version = float('.'.join(f[1].strip('v').split('-')))
+        return (survey, version)
+
+    def test_data_unchanged(self):
```

Comment:
I also fall into the habit of test functions that iterate over conditions, but it is clearer in the test suite output when you write one method to test the condition (e.g., `assertDataUnchanged(self, path)` in a following comment) and additional dedicated methods that call `assertDataUnchanged` for each survey / version.